recipes/qt/all/conanfile.py

```diff
@@ -0,0 +1,676 @@
+                                            "You can either disable qt:widgets or enable qt:GUI")
+        if not self.options.GUI:
+            del self.options.opengl
+            del self.options.with_vulkan
+            del self.options.with_freetype
+            del self.options.with_fontconfig
+            del self.options.with_harfbuzz
+            del self.options.with_libjpeg
+            del self.options.with_libpng
+
+        if not self.options.qtmultimedia:
+            del self.options.with_libalsa
+            del self.options.with_openal
+
+        if self.options.qtwebengine:
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("Static builds of Qt Webengine are not supported")
+
+            if tools.cross_building(self.settings, skip_x64_x86=True):
+                raise ConanInvalidConfiguration("Cross compiling Qt WebEngine is not supported")
+
+            if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
+                raise ConanInvalidConfiguration("Compiling Qt WebEngine with gcc < 5 is not supported")
+
+        if self.settings.os == "Android" and self.options.get_safe("opengl", "no") == "desktop":
+            raise ConanInvalidConfiguration("OpenGL desktop is not supported on Android. Consider using OpenGL es2")
+
+        if self.settings.os != "Windows" and self.options.get_safe("opengl", "no") == "dynamic":
+            raise ConanInvalidConfiguration("Dynamic OpenGL is supported only on Windows.")
+
+        if self.options.get_safe("with_fontconfig", False) and not self.options.get_safe("with_freetype", False):
+            raise ConanInvalidConfiguration("with_fontconfig cannot be enabled if with_freetype is disabled.")
+
+        if self.options.multiconfiguration:
+            del self.settings.build_type
+
+        if not self.options.with_doubleconversion and str(self.settings.compiler.libcxx) != "libc++":
+            raise ConanInvalidConfiguration('Qt without libc++ needs qt:with_doubleconversion. '
+                                            'Either enable qt:with_doubleconversion or switch to libc++')
+
+        if tools.os_info.is_linux:
+            if self.options.qtwebengine:
+                self.options.with_fontconfig = True
+
+        if "MT" in self.settings.get_safe("compiler.runtime", default="") and self.options.shared:
+            raise ConanInvalidConfiguration("Qt cannot be built as shared library with static runtime")
+
+        def _enablemodule(mod):
+            if mod != 'qtbase':
+                setattr(self.options, mod, True)
+            for req in self._submodules[mod]["depends"]:
+                _enablemodule(req)
+
+        for module in self._submodules:
+            if module != 'qtbase' and getattr(self.options, module):
+                _enablemodule(module)
+
+    def requirements(self):
```

Comment:
I don't see any conditional requirements based on qt modules, except for qtbase and qtwebengine. Several others modules depend on vendored libraries.

For example:
- qtmultimedia depends on libtiff and libwebp
- qt3d depends on assimp and imgui
- qtlocation depends on clip2tri, clipper, earcut, mapbox-gl-native, poly2tri.

Reply:
At the end of configuration:

```
Further Image Formats:

  JasPer ................................. no

  MNG .................................... no

  TIFF ................................... yes

    Using system libtiff ................. no

  WEBP ................................... yes

    Using system libwebp ................. no
```